flutter logs failed for a kitkat device

When running running flutter logs for an android emulator, api 19, kitkat (4.4.2). It looks like that API level doesn't support the `-T` option for logcat.

```
flutter logs
Showing Nexus S KitKat logs:
logcat: invalid option -- T
Unrecognized Option
Usage: logcat [options] [filterspecs]
options include:
  -s              Set default filter to silent.
                  Like specifying filterspec '*:s'
  -f <filename>   Log to file. Default to stdout
  -r [<kbytes>]   Rotate log every kbytes. (16 if unspecified). Requires -f
  -n <count>      Sets max number of rotated logs to <count>, default 4
  -v <format>     Sets the log print format, where <format> is one of:

                  brief process tag thread raw time threadtime long

  -c              clear (flush) the entire log and exit
  -d              dump the log and then exit (don't block)
  -t <count>      print only the most recent <count> lines (implies -d)
  -g              get the size of the log's ring buffer and exit
  -b <buffer>     Request alternate ring buffer, 'main', 'system', 'radio'
                  or 'events'. Multiple -b parameters are allowed and the
                  results are interleaved. The default is -b main -b system.
  -B              output the log in binary
filterspecs are a series of 
  <tag>[:priority]

where <tag> is a log component tag (or * for all) and priority is:
  V    Verbose
  D    Debug
  I    Info
  W    Warn
  E    Error
  F    Fatal
  S    Silent (supress all output)

'*' means '*:d' and <tag> by itself means <tag>:v

If not specified on the commandline, filterspec is set from ANDROID_LOG_TAGS.
If no filterspec is found, filter defaults to '*:I'

If not specified with -v, format is set from ANDROID_PRINTF_LOG
or defaults to "brief"
```

--- a/packages/flutter_tools/lib/src/android/android_device.dart
+++ b/packages/flutter_tools/lib/src/android/android_device.dart
@@ -640,8 +640,20 @@ class _AdbLogReader extends DeviceLogReader {
     // Start the adb logcat process.
     List<String> args = <String>['logcat', '-v', 'tag'];
     String lastTimestamp = device.lastLogcatTimestamp;
-    if (lastTimestamp != null)
-      args.addAll(<String>['-T', lastTimestamp]);
+    if (lastTimestamp != null) {
+      bool supportsLastTimestamp = false;
+
+      // Check to see if this copy of adb supports -T.
+      try {
+        // "logcat: invalid option -- T", "Unrecognized Option"
+        // logcat -g will finish immediately; it will print an error to stdout if -T isn't supported.
+        String result = runSync(device.adbCommandForDevice(<String>['logcat', '-g', '-T', lastTimestamp]));
+        supportsLastTimestamp = !result.contains('logcat: invalid option') && !result.contains('Unrecognized Option');
+      } catch (_) { }
+
+      if (supportsLastTimestamp)
+        args.addAll(<String>['-T', lastTimestamp]);
+    }
     runCommand(device.adbCommandForDevice(args)).then((Process process) {
       _process = process;
       _process.stdout.transform(UTF8.decoder).transform(const LineSplitter()).listen(_onLine);

